### PR TITLE
Task 1 News API

### PR DESCRIPTION
**Pull request by AlenaKharytonchyk**
**Link to ghPages:** 

https://alenakharytonchyk.github.io/FrontCamp/

Hometask:

1) Go to https://newsapi.org/
2) Press ‘Get Api Key’ on the right
3) Enter your desired creds and promise to add an attribution link to newsapi
4) Press submit and store your API key - this one will be used for api requests
5) Test you did well - https://newsapi.org/v1/articles?source=bbc-news&apiKey={{YOUR_API_KEY}}
6) Create application, using your github page on your github account: https://pages.github.com/
7) Using es6 knowledge create an application that uses newsapi, which will run purely on the client-side in Chrome-54 browser (no server-side work expected). Your app should get all the news in the available section and display them;
8) Score points for every usage of the es6, but points will be descored for prehistoric things (such as XMLHttpRequest);
9) Styling is not the requirement for this task, but it will be an additional bonus;
10) You're not allowed to use any framework :)
11) Add attribution link, remember, you promised!

Rate Criterias

5 star: Application is usable, designed well, looks good also on mobile, uses as much es6 features as possible (using es6 also in all cases, where are es5 alternatives). 
4 star: Not all cases use es6 features, designs have issues, limited responsive (looks bad on mobile).
3 star: Application has no designs at all, bad UX, only few es6 features used. 
2 star: Application has no designs, does not use es6 at all. 
1 star: Application is not working.
0 star: No application

### CODE REVIEW

File: .idea/FrontCamp.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (NikitaHapeyeu) [inline .idea/FrontCamp.iml]: Please remove idea files

File: .idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (NikitaHapeyeu) [inline .idea/misc.xml]: Please remove idea files

File: .idea/modules.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (NikitaHapeyeu) [inline .idea/modules.xml]: Please remove idea files

File: .idea/vcs.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (NikitaHapeyeu) [inline .idea/vcs.xml]: Please remove idea files

File: package.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "webpack-dev-server --config ./webpack.config.js --mode development",
+    "build": "webpack --mode production",
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d dist",
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "lint": "eslint src/**/*.js",
```
Comment (NikitaHapeyeu) [inline package.json]: Do you run the linter?

File: src/Components/app/NewsApp.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  async render() {
+    const buttonsContainer = document.querySelector('.app .buttons-container');
+    const newsContainer = document.querySelector('.app .news-container');
+    const backBtn = buttonsContainer.querySelector('.back-btn');
+
+    if (!this.source) {
+      buttonsContainer.classList.remove('display-news');
+      newsContainer.style.display = 'none';
+      newsContainer.innerHTML = '';
+      if (!this.sources) {
```
Comment (NikitaHapeyeu) [inline src/Components/app/NewsApp.js]: Duplicated logic with 24 line

File: src/Components/app/NewsApp.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    mainContainer.innerHTML = template;
+  }
+
+  setSource(source) {
+    this.source = source;
+    this.render();
+  }
+
+  async render() {
+    const buttonsContainer = document.querySelector('.app .buttons-container');
```
Comment (NikitaHapeyeu) [inline src/Components/app/NewsApp.js]: What can happen if the document isn't found in the document? Is it worth considering such a variation?

File: .idea/misc.xml (hunk 2 above)
Comment (AlenaKharytonchyk) [reply .idea/misc.xml]: removed

File: .idea/vcs.xml (hunk 4 above)
Comment (AlenaKharytonchyk) [reply .idea/vcs.xml]: removed

File: .idea/modules.xml (hunk 3 above)
Comment (AlenaKharytonchyk) [reply .idea/modules.xml]: removed

File: package.json (hunk 5 above)
Comment (AlenaKharytonchyk) [reply package.json]: image
It's working

File: src/Components/app/NewsApp.js (hunk 6 above)
Comment (AlenaKharytonchyk) [reply src/Components/app/NewsApp.js]: Done

File: src/Components/app/NewsApp.js (hunk 7 above)
Comment (AlenaKharytonchyk) [reply src/Components/app/NewsApp.js]: done

**Pull request merged**